cmake/multi_image.cmake

```
@@ -0,0 +1,164 @@
+    message("Changed board to secure nrf9160_pca20035 (NOT NS)")
+  endif()
+
+  if(${BOARD} STREQUAL actinius_icarus_ns)
+    set(BOARD actinius_icarus)
+    message("Changed board to secure actinius_icarus (NOT NS)")
+  endif()
+endfunction()
+
+function(image_menuconfig target_out command_out)
+  if(EXTRA_KCONFIG_TARGETS)
+    set(${target_out}  "-DEXTRA_KCONFIG_TARGETS=${EXTRA_KCONFIG_TARGETS}" PARENT_SCOPE)
+    set(${command_out} "-DEXTRA_KCONFIG_TARGET_COMMAND_FOR_${EXTRA_KCONFIG_TARGETS}=${EXTRA_KCONFIG_TARGET_COMMAND_FOR_${EXTRA_KCONFIG_TARGETS}}" PARENT_SCOPE)
+  endif()
+endfunction()
+
+function(zephyr_add_external_image name sourcedir)
+  string(TOUPPER ${name} UPNAME)
+
+  if (CONFIG_${UPNAME}_BUILD_STRATEGY_USE_HEX_FILE)
+    assert_exists(CONFIG_${UPNAME}_HEX_FILE)
+    message("Using ${CONFIG_${UPNAME}_HEX_FILE} instead of building ${name}")
+  elseif (CONFIG_${UPNAME}_BUILD_STRATEGY_SKIP_BUILD)
+    message("Skipping building of ${name}")
+  else()
+    # Build normally
+    zephyr_add_external_image_from_source(${name} ${sourcedir})
+  endif()
+endfunction()
+
+function(zephyr_add_external_image_from_source name sourcedir)
+  string(TOUPPER ${name} UPNAME)
+  set(${UPNAME}_CMAKE_BINARY_DIR ${CMAKE_BINARY_DIR}/${name})
+  set(${UPNAME}_PROJECT_BINARY_DIR ${CMAKE_BINARY_DIR}/${name}/zephyr)
+
+  get_cmake_property(VARIABLES_CACHED CACHE_VARIABLES)
+  get_cmake_property(VARIABLES              VARIABLES)
+
+  unset(image_cmake_args)
+  foreach(var_name
+      ${VARIABLES}
+      ${VARIABLES_CACHED}
+      )
+    if("${var_name}" MATCHES "^${name}_(.+)") # TODO: How expensive?
+      list(APPEND image_cmake_args
+        -D${CMAKE_MATCH_1}=${${var_name}}
+        )
+    endif()
+  endforeach()
+
+  file(MAKE_DIRECTORY ${${UPNAME}_CMAKE_BINARY_DIR})
+  message("=== child image ${name} begin ===")
+  image_board_selection(${BOARD} IMAGE_BOARD)
+  image_menuconfig(IMAGE_KCONFIG_TARGET IMAGE_COMMAND_TARGET)
+  execute_process(
+    COMMAND ${CMAKE_COMMAND}
+    # Add other toolchain here,
+    -G${CMAKE_GENERATOR}
+    -DBOARD=${IMAGE_BOARD}
+    ${IMAGE_KCONFIG_TARGET}
+    ${IMAGE_COMMAND_TARGET}
+    -DIS_CHILD_IMAGE=True
+    ${sourcedir}
+    ${image_cmake_args}
+    WORKING_DIRECTORY ${${UPNAME}_CMAKE_BINARY_DIR}
+    RESULT_VARIABLE ret
+    )
+  set(${UPNAME}_DOTCONFIG ${${UPNAME}_PROJECT_BINARY_DIR}/.config)
+  set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${${UPNAME}_DOTCONFIG})
+
+  if(NOT ${ret} EQUAL "0")
+    message(FATAL_ERROR "CMake generation for ${name} failed, aborting. Command: ${ret}")
+  endif()
+
+  message("=== child image ${name} end ===")
+
+  # The variables inhereted from PARENT_SCOPE might be overwritten when additional Kconfig is loaded locally.
+  # Therefore, safe guard the variables from parent scope that are needed.
+  import_kconfig(CONFIG_ ${${UPNAME}_PROJECT_BINARY_DIR}/.config)
+
+  # Clear it, just in case it was defined in parent scope
+  set(spm_veneers_lib)
```

Today, only the `spm` provides secure entries, but that could change.
So I suggest to decouple the spm name from the veneers lib.
```suggestion
  set(secure_veneers_lib)
```

---

will look at this later